src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ProvideStreamMemoryAsyncOverridesTests.cs

```diff
@@ -0,0 +1,1476 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Testing;
+using Test.Utilities;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ProvideStreamMemoryBasedAsyncOverrides,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.ProvideStreamMemoryBasedAsyncOverrides,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class ProvideStreamMemoryBasedAsyncOverridesTests
+    {
+        #region Reports Diagnostic
+        [Fact]
+        public Task ReadAsyncArray_NoReadAsyncMemory_ReportsDiagnostic_CS()
+        {
+            string code = $@"
+{CSUsings}
+namespace Testopolis
+{{
+    public class {{|#0:FooStream|}} : Stream
+    {{
+        {CSAbstractMembers}
+        {CSReadAsyncArray}
+    }}
+}}";
```

Comment:
[cosmetic] I am fine if you keep it this way but I find the double curly braces hard to read.

```suggestion
            string code = @"
" + CSUsings + @"
namespace Testopolis
{
    public class {|#0:FooStream|} : Stream
    {
        " + CSAbstractMembers + @"
        " + CSReadAsyncArray + @"
    }
}";
```

Reply:
Going forward, I will do it your way.

Reply:
Let's wait for @sharwell or @mavasani opinion here.